### PR TITLE
WT-13482 Eviction module poc

### PR DESCRIPTION
**Pull request by keitharnoldsmith**
This branch is POC code to test/demonstrate building a clearer API boundary for eviction. 

PR is mostly for visibility plus automatic testing as changes merge.

### CODE REVIEW

File: dist/filelist [hunk 1]
```diff
@@ -80,6 +80,7 @@ src/conn/conn_chunkcache.c
 src/conn/conn_ckpt.c
 src/conn/conn_compact.c
 src/conn/conn_dhandle.c
+src/conn/conn_evict.c
```
Comment (sueloverso) [inline dist/filelist:83]: Creating this gives us both `conn_cache.c` and `conn_evict.c`. Eviction is an action taken to service the cache. Do you intend to remove `conn_cache.c`? Or, at least to me, the "in which file do I put something new" question becomes potentially harder to reason about.

File: src/conn/conn_evict.c [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+            WT_RET_MSG(session, EINVAL, "%s should not exceed cache size", param_name);
+
+        *param = (input * 100.0) / (conn->cache_size);
+    }
+
+    return (0);
+}
+
+/*
+ * __validate_evict_config --
```
Comment (sueloverso) [inline src/conn/conn_evict.c:54]: I realize it was poorly named originally and possibly not fully described. Can this be fixed to follow the convention to be something like `__evict_validate_config` or `__evict_config_validate`? But in consideration of that, the function does both the parsing and validation of those values. So the comment is  a bit misleading.

File: src/conn/conn_evict.c [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+    evict->eviction_updates_target = (double)cval.val;
+    WT_RET(__evict_config_abs_to_pct(
+      session, &(evict->eviction_updates_target), "eviction updates target", shared));
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_updates_trigger", &cval));
+    evict->eviction_updates_trigger = (double)cval.val;
+    WT_RET(__evict_config_abs_to_pct(
+      session, &(evict->eviction_updates_trigger), "eviction updates trigger", shared));
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_checkpoint_target", &cval));
```
Comment (sueloverso) [inline src/conn/conn_evict.c:108]: I realize much of this is code movement from the other file. But please alphabetize the config parsing unless there is an ordering need to do so.

File: src/conn/conn_evict.c [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+int
+__wti_eviction_config(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
+    WT_ASSERT(session, conn->evict != NULL);
+
+    WT_RET(__evict_config_local(session, cfg));
```
Comment (sueloverso) [inline src/conn/conn_evict.c:237]: The `evict_config_local` function is quite small. And this is the *only* caller of it. Why is it a function? Why not have that code in here directly given its small size?

**Review by sueloverso — COMMENTED** [review]
I realize this is early so I just made a few comments. I have a general comment from this diff as well that may apply to later/other modules and PRs.

I think the renaming/extrapolation from cache to evict structures and that usage should be a separate PR and change from code restructuring. The vast amount of renaming makes it harder to see if there are any substantive other changes. So splitting changes up will make that easier to see.

File: src/include/evict_inline.h [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+
+/*
+ * __wt_evict_page_needed --
+ *     We are using a page, so it was marked as not needed, that is no longer the case.
+ */
+static WT_INLINE void
+__wt_evict_page_needed(WT_SESSION_IMPL *session, WT_PAGE *page)
+{
+    if (__wt_atomic_load64(&page->read_gen) == WT_READGEN_WONT_NEED)
+        __wt_cache_read_gen_new(session, page);
```
Comment (Siddhartha8899) [inline src/include/evict_inline.h:485]: I think this can be made private :-)

**Pull request closed without merging**